src/wp-admin/options-reading.php

```diff
@@ -79,7 +79,7 @@
 
 else :
 	if ( 'page' === get_option( 'show_on_front' ) && ! get_option( 'page_on_front' ) && ! get_option( 'page_for_posts' ) ) {
-		update_option( 'show_on_front', 'posts' );
+		update_option( 'show_on_front', 'posts', true );
```

Comment:
https://github.com/WordPress/wordpress-develop/pull/6465#discussion_r1731897851
```suggestion
		update_option( 'show_on_front', 'posts' );
```